src/server_manager/web_app/browser_main.ts

```diff
@@ -13,8 +13,9 @@
 // limitations under the License.
 
 // eslint-disable-next-line @typescript-eslint/no-explicit-any
-(window as any).trustCertificate = (host: string, fingerprint: string) => {
+(window as any).trustCertificate = (host: string, fingerprint: string): boolean => {
```

Comment:
I feel like this is now a low level function hiding application logic. It's up to the application layer to decide what to reject.

I'd revert this and have a way for the application to validate manual entries before adding them.

Reply:
I'm not sure what you're suggesting.  Are you proposing that we add a function like `getTrustedFingerprint(host: string): string`, and let the application decide what to do if a fingerprint has been set and doesn't match?  In this case, I would want to create a new shared component on the browser side that checks for collisions before calling `trustCertificate()`, to avoid replicating the collision-prevention logic at each callsite.

I don't see the advantage of this approach.  It seems like an unsafe API that is not necessary for our use cases.